tests/integration/test_logcollector/test_command_monitoring/test_command_execution.py

```diff
@@ -0,0 +1,214 @@
+    {'log_format': 'full_command', 'command': 'テ``ñスト, ИСПЫТА´НИЕ",\'测`试", "اختبا', 'alias': ''},
+    {'log_format': 'full_command', 'command': 'echo ***', 'alias': ''},
+]
+
+if sys.platform == 'linux':
+    # Command that took few seconds and is killed by the system/test
+    parameters.append({'LOG_FORMAT': 'command',
+                       'COMMAND': 'timeout 2 tail -f /var/ossec/logs/active-responses.log', 'ALIAS': ''})
+    parameters.append({'LOG_FORMAT': 'full_command',
+                       'COMMAND': 'timeout 2 tail -f /var/ossec/logs/active-responses.log', 'ALIAS': ''})
+    # Command with many arguments.
+    parameters.append({'LOG_FORMAT': 'command', 'COMMAND': 'ss -l -p -u -t -4 -6 -n', 'ALIAS': ''})
+    parameters.append({'LOG_FORMAT': 'full_command', 'COMMAND': 'ss -l -p -u -t -4 -6 -n', 'ALIAS': ''})
+
+    metadata.append({'log_format': 'command',
+                     'command': 'timeout 2 tail -f /var/ossec/logs/active-responses.log', 'alias': ''})
+    metadata.append({'log_format': 'full_command',
+                     'command': 'timeout 2 tail -f /var/ossec/logs/active-responses.log', 'alias': ''})
+    metadata.append({'log_format': 'command',
+                     'command': 'ss -l -p -u -t -4 -6 -n', 'alias': ''})
+    metadata.append({'log_format': 'full_command',
+                     'command': 'ss -l -p -u -t -4 -6 -n', 'alias': ''})
+
+configurations = load_wazuh_configurations(configurations_path, __name__, params=parameters, metadata=metadata)
+configuration_ids = [f"{x['LOG_FORMAT'], x['COMMAND'], x['ALIAS']}" for x in parameters]
+
+
+# fixtures
+@pytest.fixture(scope="module", params=configurations, ids=configuration_ids)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture(scope="module")
+def get_local_internal_options():
+    """Get internal configuration."""
+    return local_internal_options
+
+
+def dbg_reading_command(command, alias, log_format):
+    """Check if the (previously known) output of a command ("echo") is displayed correctly.
+
+    It also checks if the "alias" option is working correctly.
+
+    Args:
+        command (str): Command to be monitored.
+        alias (str): An alternate name for the command.
+        log_format (str): Format of the log to be read ("command" or "full_command").
+
+    Raises:
+        TimeoutError: If the command monitoring callback is not generated.
+    """
+    prfx = prefix
+    output = command
+    output = output.replace('echo ', '')
+
+    if log_format == 'full_command':
+        msg = fr"^{output}'"
+        prfx = ''
+    else:
+        msg = fr"DEBUG: Reading command message: 'ossec: output: '{alias}': {output}'"
+
+    wazuh_log_monitor.start(timeout=5, callback=monitoring.make_callback(pattern=msg, prefix=prfx),
+                            error_message=logcollector.GENERIC_CALLBACK_ERROR_COMMAND_MONITORING)
```

Comment:
Maybe the test should be more precise if you compare the output with the real system output using `os.system`.

Reply:
Done in https://github.com/wazuh/wazuh-qa/pull/1257/commits/1e33fb8c458aeb8a738e22706713ec98fbf25703